### ISSUE TITLE
Add a "Test" button to Program Configuration Pages

### ISSUE THREAD

**Issue by dalgibbard** [MediaFrontPage/mediafrontpage#19]
When adding a Program using the "Settings> Programs> [ProgramIcon]" option, it would be extraordinarily useful to have a button labeled "Test" which just does a simple connect to the remote application and validates that it receives the output it expects - this would be useful for validating that:
- MFP can connect to the application
- That the authentication details are correct
- Returned data from the application is as expected.

This should save time for users who are unsure on how to configure certain aspects, and enable them to run a series of probes for different configurations quickly, rather than having to go back to the main page to see if things are working or not.

**Comment by gugahoi** [comment]
Wow, this is a pretty old project. It hasn't been updated in a very very long time. I am happy that someone found it useful tho. For the most part of the last couple of years I have migrated to a different project called Maraschino -- https://github.com/mrkipling/maraschino. I highly advise that you check it out.

**Comment by dalgibbard** [comment]
Yeah I realised that pretty much immediately after posting :) never mind!
Thanks.

**Comment by MikeEsteves** [comment]
There are some users who are maintaining the project. Look at the different
forks and one of them has new features.

**Comment by DejaVu** [comment]
Interesting Commit name!! 
https://github.com/DivinityCycle/mediafrontpage/commit/6f1aaf67595fd9f2369e9e4dbbabd377f7e27cb2

One of the main reasons it was abandoned if I recall! :)
